### ISSUE TITLE
Jobs with the same underlying graph are duplicated in Graphs tab in Dagit

### ISSUE THREAD

**Issue by dinya** [dagster-io/dagster#5412]
Version 0.13.1 (jobs/graphs/ops as recommended primitives). Graph and jobs are created with

```python
@graph(name="process_records_graph")
def process_records_graph():
    process_records()

process_records_job = process_records_graph.to_job(
    name="do_process_records",
    config=cfg1
)

process_records_job2 = process_records_graph.to_job(
    name="do_process_records2",
    config=cfg2
)
```

Dagit UI:

1, Jobs tab:
   ![0 jobs_tab](https://user-images.githubusercontent.com/2182222/139022400-fbce5f9b-7f23-456b-80d7-ce2e0cec4808.jpg)
2. Graphs tab. Two elements ``process_records_job`` and ``process_records_job2`` (named the same as "forked" jobs with the same contents):
   ![1 graphs_tab](https://user-images.githubusercontent.com/2182222/139022404-e5c0891a-471b-43c7-89d1-9be4c057f455.jpg)
   ![2 graph_do_process_records](https://user-images.githubusercontent.com/2182222/139022407-7f5ad01b-bb2b-4969-a924-5eb0603bd544.jpg)
   ![3 graph_do_process_records2](https://user-images.githubusercontent.com/2182222/139022408-67882166-5e8a-4f0d-a1df-4644b8764d31.jpg)
3. Expected view is something like (just one element ``process_records_graph`` ):
  ![1 1 graphs_tab (expected)](https://user-images.githubusercontent.com/2182222/139022406-e9a3b32d-7309-4b05-8c84-35773d3fc3a1.jpg)

According to discussion in [dagster slack](https://app.slack.com/client/TCDGQDUKF/C01U954MEER/thread/C01U954MEER-1635151863.077000) with @dpeng817 this behavior looks like a bug.

**Comment by dinya** [comment]
Version 0.13.4 looks like partially solving the issue

1. Graph tab is ok (looks as expected)
 
![dagit-0 13 4-graph-tab](https://user-images.githubusercontent.com/2182222/140505790-00a256f6-e7a6-4b92-99c7-21dff17f7405.png)

2. But after clicking on ``process_records_graph``

![dagit-0 13 4-graph-tab-fails](https://user-images.githubusercontent.com/2182222/140505780-caf50bed-fabd-4000-86b9-934e4bb8908b.png)

**Comment by dpeng817** [comment]
cc @prha

**Comment by prha** [comment]
I believe this is fixed now in `0.13.5`
